extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -2304,4 +2272,72 @@ public static function onGetTalkPage( Title $title, Title &$talkPageTitle ) {
 		return true;
 	}
 
+	/**
+	 * SUS-845: Lock down Thread namespaces and Message Wall Greetings
+	 * For Thread namespaces, pnly allow edits to users with proper permission or to people editing their own posts
+	 * Everything else is disallowed.
+	 * For Wall greetings, only allow edits by owner or users with proper permission
+	 * @param Title $title
+	 * @param User $user
+	 * @param string $action
+	 * @param bool|string|string[] $result true or string or array of strings (MediaWiki error message key(s) to be shown if error)
+	 * @return bool Whether the user is allowed to perform this action or true if this is not a Wall/Forum title
+	 */
+	public static function onGetUserPermissionsErrors( Title $title, User $user, string $action, &$result ): bool {
+		global $wgWallNS, $wgIsSafeWallTransaction;
+
+		// Wall or Forum threads
+		if ( in_array( MWNamespace::getSubject( $title->getNamespace() ), $wgWallNS ) ) {
+			$wallMessage = new WallMessage( $title );
+			$isAuthor = $wallMessage instanceof WallMessage && $wallMessage->isAuthor( $user );
+			$isActionAllowed = true;
+			switch ( $action ) {
+				case 'edit':
+					$isActionAllowed = $isAuthor || $user->isAllowed( 'walledit' );
+					$result = $isActionAllowed ?? [ 'badaccess-group0' ];
+					break;
+				case 'create':
+					// Pages in Thread or Board Thread namespaces should only be created via Nirvana API
+					// Manually creating one results in a broken page and broken RC entry
+					$isActionAllowed = $wgIsSafeWallTransaction ?? false;
```

Comment:
Although in this case using the null coalescing operator is correct, in my opinion `$isActionAllowed = $wgIsSafeWallTransation || false` is actually a good use of weak typing in PHP as it will make `$isActionAllowed` Boolean. Which in turn makes the above assignment unnecessary so I'd suggest changing the `if` below to `if ( $wgIsSafeWallTransaction )` which is readable and easy to understand.